### DISCUSSION TITLE
Estimated date for a stable 2.x release

### DISCUSSION THREAD

**Discussion by klaymond** [viewflow/viewflow#407] [General]
Hello viewflow team,  
First of all let me tell you I love the project and the new 2.x API and UI. The first major release has been on my radar for some years but I never got a use case that really fit. I will soon be pitching viewflow as a solution for a client (finally) and I'm excited that it will possibly be using 2.x.  
  
I have run into some minor issues with the beta versions. I have built a POC that circumvents these issues and the lack of Pro license. Before I go through with my presentation of the POC I would like to know what the estimated dates for a stable version are so I can let the clients know of a timeline for MVP and production. For this project I would be using the lowest tier license for this project but I am looking to sell this solution to more potential customers if things go right so the high tier license is on the horizon. I am excited to see what the Pro nodes are and especially looking forward to using celery jobs in my flows.  
  
TL;DR  
What are the estimated dates for release of a stable viewflow 2.x Pro?  
  
Best regards,  
Uri G

**Comment by klaymond** [comment]
Hello. Just wondering if there is an answer to my question.

**Comment by kmmbvnr** [comment]
There is no specific date yet, for the stable 2.x release

**Comment by marcfargas** [comment]
@kmmbvnr  via https://github.com/viewflow/django-fsm/issues/298#issuecomment-1935567038 :

> Guys, Viewflow would be released soon as a stable.
>
> viewflow.fsm is the django-fsm (3.0)
>
> django-fsm 2.x was released as 2014 has several design flaws and quite obsolete

**Comment by marcfargas** [comment]
@kmmbvnr I am reviewing `viewflow` for a project for the first time but I have some... concerns?

1. It looks like viewflow is, and has always been, a solo project. It it so?
2. 2.0 development seems to have stalled somehow. There are commits on this repo, but the pypi release cadence has stalled (in 2023 there was a release for almost each month until September and nothing since then)
3. For Pro the website advertises "private e-mail support" but no SLA. What is the average reply time on that?
4. For Pro, is there also an equivalent Pro v2 beta like on the Open Source version or is it still v1?

Do you have any plans regarding a possible future project abandonment like "if I abandon viewflow I will release the source code as MIT license" or something like that?

Thank you,
marc

**Comment by kmmbvnr** [comment]
@telenieko I understand your concerns regarding the sustainability of the project. As a note, I would like to point you to the changelog that shows Django-FSM/Django-Viewflow has been constantly developed since 2010.The commits in the open-source part show that it is managed by occasional syncs from the pro version.For inquiries about the pro version, please contact mikhail@viewflow.io

**Reply by marcfargas** [reply]
Thank you for the prompt reply!

**Comment by ShaheedHaque** [comment]
Is there a summary of the API changes required by clients to migrate between 1.11 and 2.0?

**Comment by kmmbvnr** [comment]
It's quite big, starter migration guide available in wiki - https://github.com/viewflow/viewflow/wiki/Django%E2%80%90Material-and-Viewflow-1.x-migration-Guide
